core/src/main/java/org/axonframework/commandhandling/model/inspection/AnnotatedAggregate.java

```diff
@@ -65,59 +71,128 @@
      * Initialize an Aggregate instance for the given {@code aggregateRoot}, described by the given
      * {@code aggregateModel} that will publish events to the given {@code eventBus}.
      *
-     * @param aggregateRoot The aggregate root instance
-     * @param model         The model describing the aggregate structure
-     * @param eventBus      The Event Bus to publish generated events on
+     * @param aggregateRoot      The aggregate root instance
+     * @param model              The model describing the aggregate structure
+     * @param eventBus           The Event Bus to publish generated events on
      */
     protected AnnotatedAggregate(T aggregateRoot, AggregateModel<T> model, EventBus eventBus) {
+        this(aggregateRoot, model, eventBus, null);
+    }
+
+    /**
+     * Initialize an Aggregate instance for the given {@code aggregateRoot}, described by the given
+     * {@code aggregateModel} that will publish events to the given {@code eventBus}.
+     *
+     * @param aggregateRoot      The aggregate root instance
+     * @param model              The model describing the aggregate structure
+     * @param eventBus           The Event Bus to publish generated events on
+     * @param repositoryProvider Provides repositories for given aggregate types
```

Comment:
What are the 'given aggregate types' here? Can it provide any Aggregate Repo, or a specified set?
I would guess the latter is the case, which thus suggests adjusting the javadoc to `Provides repositories for specific aggregate types`.